python/paddle/fluid/dygraph/varbase_patch_methods.py

```diff
@@ -1094,7 +1094,7 @@ def dtype_str(dtype):
                 # for example, paddle.fluid.core.VarDesc.VarType.LOD_TENSOR
                 return origin(dtype)
 
-        setattr(core.VarDesc.VarType, "__repr__", dtype_str)
```

Comment:
替换成__str__的原因是?

Reply:
使用 `str(x.dtype)` 输出 Tensor 的 dtype 时，实际调用的是 `__str__` 而非 `__repr__` (Ref. [math.py line2872](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/tensor/math.py#L2872))

Actually, it calls `__str__` instead of `__repr__` (Ref. [math.py line2872](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/tensor/math.py#L2872)) when outputting Tensor's dtype.